crates/test-utils/src/util.rs

```diff
@@ -51,30 +53,62 @@ pub fn template_lock() -> RwLock<File> {
     RwLock::new(lock_file)
 }
 
-/// Copies an initialized project to the given path
+/// Initializes a project with `forge init` at the given path.
+///
+/// This should be called after an empty project is created like in
+/// [some of this crate's macros](crate::forgetest_init).
+///
+/// ## Note
+///
+/// This doesn't always run `forge init`, instead opting to copy an already-initialized template
+/// project from a global template path. This is done to speed up tests.
+///
+/// This used to use a `static` [`Lazy`], but this approach does not with `cargo-nextest` because it
+/// runs each test in a separate process. Instead, we use a global lock file to ensure that only one
+/// test can initialize the template at a time.
 pub fn initialize(target: &Path) {
-    eprintln!("initialize {}", target.display());
+    eprintln!("initializing {}", target.display());
 
-    let tpath = &*TEMPLATE_PATH;
+    let tpath = TEMPLATE_PATH.as_path();
     pretty_err(tpath, fs::create_dir_all(tpath));
 
+    // Initialize the global template if necessary.
     let mut lock = template_lock();
-    let read = lock.read().unwrap();
+    let mut _read = Some(lock.read().unwrap());
     if fs::read_to_string(&*TEMPLATE_LOCK).unwrap() != "1" {
-        eprintln!("initializing template dir");
-
-        drop(read);
+        // We are the first to acquire the lock:
+        // - initialize a new empty temp project;
+        // - run `forge init`;
+        // - run `forge build`;
+        // - copy it over to the global template;
+        // Ideally we would be able to initialize a temp project directly in the global template,
+        // but `TempProject` does not currently allow this: https://github.com/foundry-rs/compilers/issues/22
+
+        // Release the read lock and acquire a write lock, initializing the lock file.
```

Comment:
ah yeah we should have thought about this much earlier lol

Reply:
The problem is that the static Lazy didn't work due to nextest running each test in a separate process